percona/controllers/pgc/pgc.go

```diff
@@ -113,8 +113,26 @@ func (c *Controller) onAdd(obj interface{}) {
 
 	c.Queue.Add(key)
 	defer c.Queue.Done(key)
+
 	newCluster := obj.(*crv1.PerconaPGCluster)
+	nn := types.NamespacedName{
+		Name:      newCluster.Name,
+		Namespace: newCluster.Namespace,
+	}
+	if c.lockers.store == nil {
+		c.lockers = newLockStore()
+	}
 
+	l := c.lockers.LoadOrCreate(nn.String())
+	l.statusMutex.Lock()
+	defer l.statusMutex.Unlock()
+
+	err = version.EnsureVersion(c.Client, newCluster, version.VersionServiceClient{
+		OpVersion: newCluster.ObjectMeta.Labels["pgo-version"],
+	})
+	if err != nil {
+		log.Errorf("ensure version %s", err)
```

Comment:
why we're continuing execution if error?

Reply:
https://github.com/percona/percona-postgresql-operator/pull/181/commits/ed0fd46c2efa1c89d88cd96a34961a807fa40913